scripts/release_level.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+# Script to determine the next release level based on conventional commits.
+#
+# Possible levels:
+# * patch
+# * minor
+#
+# Any feature commits indicates a minor release. The lack of feature commits indicates
+# a patch release. Major releases are expected to be handled manually.
+#
+# Assumptions:
+# * git is installed
+# * git-cliff is installed
+# * grep is installed
+
+# Ensure we are in the git root
+cd $(git rev-parse --show-toplevel)
+
+level=patch
+
+# Using git cliff determine if there are any feat commits that do not belong to a tag.
+git cliff --unreleased --strip all --body "{% for group, commits in commits | group_by(attribute=\"group\") %}
+{{ group }} {{ commits | length }}
+{% endfor %}" | grep Features
+ret=$?
+if [[ $ret = 0 ]]
+then
+    level=minor
```

Comment:
When/how would a major release be made?

Reply:
In the future we would detect breaking changes (i.e `feat!: new breaking feature`)  and and mark it as a major change. However since we are pre 1.0 release breaking changes don't require a major change. 

So I figure we can update this script once we have our first 1.0 release.

Reply:
There are lots of things we will want to do once we have a 1.0 release, like use  https://github.com/rust-lang/rust-semverver to ensure the API isn't changing in a minor release.

Reply:
Oh, cool, makes sense.